src/reflowscan.cpp

```diff
@@ -0,0 +1,187 @@
+/*
+Minetest
+Copyright (C) 2016 MillersMan <millersman@users.noreply.github.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "reflowscan.h"
+#include "map.h"
+#include "mapblock.h"
+#include "nodedef.h"
+#include "util/timetaker.h"
+
+
+ReflowScan::ReflowScan(Map *map, INodeDefManager *ndef) :
+	m_map(map),
+	m_ndef(ndef),
+	m_liquid_queue(NULL)
+{
+}
+
+void ReflowScan::scan(MapBlock * block, UniqueQueue<v3s16> *liquid_queue)
+{
+	m_block_pos = block->getPos();
+	m_rel_block_pos = block->getPosRelative();
+	m_liquid_queue = liquid_queue;
+
+	memset(m_lookup, 0, sizeof(m_lookup));
+	int block_idx = 1 + (1 * 9) + (1 * 3);
+	m_lookup[block_idx] = block;
+	m_lookup_state_bitset = 1 << block_idx;
+
+	// Scan the columns in the block
+	for (s16 z = 0; z < MAP_BLOCKSIZE; z++)
+	for (s16 x = 0; x < MAP_BLOCKSIZE; x++) {
+		scanColumn(x, z);
+	}
+
+	// Scan neighbouring columns
+	for (s16 i = 0; i < MAP_BLOCKSIZE; i++) {
+		scanColumn(i, -1);
+		scanColumn(i, MAP_BLOCKSIZE);
+		scanColumn(-1, i);
+		scanColumn(MAP_BLOCKSIZE, i);
+	}
+}
+
+inline MapBlock *ReflowScan::lookupBlock(int x, int y, int z)
```

Comment:
The inline keyword is just a hint to the compiler that it doesn't have to follow. I've added it because inlining might allow optimizations like merging division and modulo (ok, they likely already got replaced with shifts, as MAP_BLOCKSIZE is a power of two) or eliminating some operations like adding MAP_BLOCKSIZE to x/y/z twice.

In short: IMHO having inline there might bring more advantages than removing them but they aren't absolutely necessary